### PR TITLE
Print UTF-8 warning only when actually encountering non-ascii filenames.

### PR DESCRIPTION
**Pull request by jpakkane**

### CODE REVIEW

File: mesonbuild/mesonlib.py [hunk 1]
```diff
@@ -59,6 +59,32 @@ def detect_meson_py_location():
...
+        return True
+
+def check_direntry_issues(direntry_array):
+    import locale
+    # Warn if the locale is not UTF-8. This can cause various unfixable issues
+    # such as os.stat not being able to decode filenames with unicode in them.
+    # There is no way to reset both the preferred encoding and the filesystem
+    # encoding, so we can just warn about it.
+    e = locale.getpreferredencoding()
+    if e.upper() != 'UTF-8' and not mesonlib.is_windows():
```
Comment (ignatenkobrain) [inline mesonbuild/mesonlib.py]: __[Flake8]__

undefined name 'mesonlib'

File: mesonbuild/mesonlib.py [hunk 2]
```diff
@@ -59,6 +59,32 @@ def detect_meson_py_location():
...
+        if not isinstance(direntry_array, list):
+            direntry_array = [direntry_array]
+        for de in direntry_array:
+            if is_ascii_string(de):
+                continue
+            mlog.warning('''You are using {!r} which is not a Unicode-compatible '
```
Comment (ignatenkobrain) [inline mesonbuild/mesonlib.py]: __[Flake8]__

undefined name 'mlog'

**Comment by nirbheek** [comment]
CI failure is because you're decoding string objects instead of bytes objects.

File: mesonbuild/mesonlib.py [hunk 3]
```diff
@@ -59,6 +60,35 @@ def detect_meson_py_location():
     python_command = [sys.executable]
     meson_command = python_command + [detect_meson_py_location()]
 
+def is_ascii_string(astring):
+    try:
+        if isinstance(astring, str):
+            string.encode('ascii')
```
Comment (ignatenkobrain) [inline mesonbuild/mesonlib.py]: __[Flake8]__

undefined name 'string'

File: mesonbuild/mesonlib.py [hunk 4]
```diff
@@ -59,6 +60,35 @@ def detect_meson_py_location():
...
+        else:
+            string.decode('ascii')
```
Comment (ignatenkobrain) [inline mesonbuild/mesonlib.py]: __[Flake8]__

undefined name 'string'

**Pull request merged**